src/App.js

```diff
@@ -0,0 +1,210 @@
+      case LOTTOS_ACTION.SET_LOTTO_MANUAL_TICKET:
+        return this.setLottoManualTicket(prevState, data);
+      case LOTTOS_ACTION.TOGGLE_LOTTO_DISPLAY:
+        return this.toggleLottoDispaly(prevState, data);
+      case LOTTOS_ACTION.SHOW_LOTTO_RESULT:
+        return this.updateLottoResult(prevState, data);
+      case LOTTOS_ACTION.CLOSE_MODAL:
+        return this.closeResultModal(prevState);
+      case LOTTOS_ACTION.RESTART:
+        return this.restart();
+      default:
+        prevState;
+    }
+  };
+
+  buyLottos = (prevState, purchaseMoney) => {
+    const lottos = new Array(purchaseMoney / 1000).fill([]).map(() => createRandomLotto(LOTTO_MIN, LOTTO_MAX, 6));
+    return [
+      {
+        ...prevState,
+        lottos,
+        purchaseMoney,
+      },
+      ["lottoTicket", "lottoWinningNumber"],
+    ];
+  };
+
+  getManualLottos = (lottoNums) => {
+    const lottos = [];
+    let temp = [];
+    for (let i = 0; i < lottoNums.length; i++) {
+      temp.push(lottoNums[i]);
+      if (temp.length === 6) {
+        lottos.push(temp);
+        temp = [];
+      }
+    }
+    return lottos;
+  };
+
+  buyManualLottos = (prevState, lottoNums) => {
+    const { manualTicket } = prevState;
+    const purchaseMoney = Number($(".input-purchase").value);
+    const randomLottos = new Array(purchaseMoney / 1000 - manualTicket)
+      .fill([])
+      .map(() => createRandomLotto(LOTTO_MIN, LOTTO_MAX, 3));
+
+    return [
+      {
+        ...prevState,
+        purchaseMoney: purchaseMoney,
+        manualTicket: 0,
+        lottos: [...this.getManualLottos(lottoNums), ...randomLottos],
+      },
+      ["lottoManualTicket", "lottoTicket", "lottoWinningNumber"],
+    ];
+    
+  };
+
+  setLottoManualTicket = (prevState, manualTicket) => {
+    return [{ ...prevState, manualTicket }, ["lottoManualTicket"]];
+  };
+
+  toggleLottoDispaly = (prevState, checked) => {
+    return [{ ...prevState, toggle: checked }, ["lottoTicket"]];
+  };
+
+  updateLottoResult = (prevState, { winningNums, bonusNum }) => {
+    const { lottos, purchaseMoney } = prevState;
+    const result = this.getResult(cloneDeep(LOTTOS_RESULT), lottos, winningNums, bonusNum);
+    const prizeMoney = this.getPrizeMoney(result);
+
+    return [
+      {
+        ...prevState,
+        result,
+        winningNums,
+        bonusNum,
+        showResultModal: true,
+        prizeMoney,
+        earningRatio: this.getEarningRatio(prizeMoney, purchaseMoney),
+      },
+      ["lottoResultModal"],
+    ];
+  };
+
+  closeResultModal = (prevState) => {
+    return [{ ...prevState, showResultModal: false, toggle: false }, ["lottoResultModal"]];
+  };
+
+  restart = () => {
+    return [LOTTOS_STATE, ["lottoInput", "lottoTicket", "lottoWinningNumber", "lottoResultModal"]];
+  };
+
+  getPrizeMoney = (lottoResult) => {
+    let prizeMoney = 0;
+    for (let [_, p] of Object.entries(lottoResult)) {
+      prizeMoney += p[0] * p[1];
```

Comment:
변수에 다시 할당하고 계산하거나 처음에 `lottoResult`에서 객체 형태로 데이터를 받아봐도 좋을 것 같아요.
`p[0]`과 `p[1]`의 맥락을 파악하기 위해서 `lottoResult`까지 찾아 올라가야 했어요!

```suggestion
      const winCount = p[0];
      const winPrize = p[1];
      prizeMoney += winCount * winPrize;
```

Reply:
맞네요!! 변수를 할당해서 명시적으로 하는게 더 명확할꺼 같네요